Add cap staging target for rails 6.0

ianheggie-oaf opened the pull request:
## Relevant issue(s)

* https://github.com/openaustralia/morph/issues/1412

## What does this do?

Add a staging capistrano deploy target and rails environment and update the configuration files accordingly

Also adds mailcatcher gem for development catching of emails.

## Why was this needed?

Staging needs to be different from production. It is cleanest to set RAILS_ENV=staging so stuff that shouldn't be set in staging is automatically configured accordingly

ianheggie-oaf committed:
Add staging group and dynamic inventory for provision

ianheggie-oaf committed:
Add staging cap and rails environment

ianheggie-oaf committed:
Export server_name to provisioned server to differentiate production and staging from each other

ianheggie-oaf committed:
Pick up env from .env for capistrano

ianheggie-oaf committed:
Fix accidental conflicting use of deploy_to when deploying to staging

ianheggie-oaf committed:
Describe staging setup and reformat README.md

# Conflicts:
#	README.md
#	env-example

ianheggie-oaf committed:
Add extra make targets and correct comments in playbook

ianheggie-oaf committed:
Clarify requirements for staging .env settings

# Conflicts:
#	env-example

ianheggie-oaf committed:
Use SERVER_NAME if set in production

# Conflicts:
#	config/environments/production.rb

ianheggie-oaf committed:
Clarify staging requirements in env-example

ianheggie-oaf committed:
Setup staging so it doesn't conflict with production

ianheggie-oaf committed:
Remove duplicate clean and set PATH so inventory uses .venv/bin

ianheggie-oaf committed:
set ansible github_* variables from .env via inventory script

Ensures python-dotenv library is installed

ianheggie-oaf committed:
Add the ansible helper ENV variables from Vagrantfile into Makefile

ianheggie-oaf committed:
Add missing rvm_users for staging

ianheggie-oaf committed:
Add mailcatcher for development catching of emails to match config

ianheggie-oaf committed:
Allow override of cuttlefish port

ianheggie-oaf committed:
Use a mailpit server for staging by default

ianheggie-oaf committed:
Clarify CUTTLEFISH_* settings in env

ianheggie-oaf committed:
Dont use production cuttlefish and stripe when provisioning local or remote staging

ianheggie-oaf committed:
Allow cuttlefish user and password to not be set as mailpit doesn't require it

ianheggie-oaf committed:
Add a default for STAGING_HOSTNAME in staging so asset precompile works when deploying

ianheggie-oaf committed:
Add staging env for faye

ianheggie-oaf committed:
Set rails_env to staging on remote and vagrant staging servers

ianheggie-oaf committed:
set cuttlefish_port in provision/group_vars

ianheggie-oaf committed:
Always provision a discourse secret

ianheggie-oaf committed:
Add dummy stripe details for staging and local (develpment) staging

ianheggie-oaf committed:
Flag template as yml.j2 rather than yml to stop syntax complaints in IDE

ianheggie-oaf committed:
Add myself and other oaf coders to initial discourse admin list

ianheggie-oaf committed:
fix template destination for discourse's app.yml.j2

ianheggie-oaf committed:
Use the current repo to deploy so we can deploy from personal forks to staging

Also adds 5 second pause so you can abort if warnings are issued

ianheggie-oaf committed:
fix asset:precompile complaining The production environment does not exist in .../config/sync.yml on staging server

ianheggie-oaf committed:
Set rails_env for capistrano deployment explicitly

ianheggie-oaf committed:
Fix RAILS_ENV passed to docker

ianheggie-oaf committed:
Create and use Procfile.staging for staging

ianheggie-oaf committed:
Clarify env-example and add env-staging-example

ianheggie-oaf committed:
Pass extra variables into staging from .env.staging and .env

ianheggie-oaf committed:
Pass variables into local staging (vagrant) from .env.vagrant and .env

ianheggie-oaf committed:
Document .env.vagrant usage

ianheggie-oaf committed:
Fix staging configuration from SERVER_NAME

ianheggie-oaf committed:
Move discourse_secret to group_vars/all.yml, remove ENV lookup from group_vars files (as .env may not be in ENV)

ianheggie-oaf committed:
Clarify env template

ianheggie-oaf committed:
Fix group_vars headings

ianheggie-oaf committed:
Fix rubocop complaints in Vagrantfile

Br3nda commented on config/deploy.rb:
```diff
@@ -1,24 +1,42 @@
 set :application, 'morph'
-set :repo_url, 'https://github.com/openaustralia/morph.git'
+set :repo_url, `git config --get remote.origin.url`.strip
```
ah, this means it uses whatever you have configured locally. Not sure how this is better.

Br3nda reviewed (changes requested):
can we keep it pointing at the OAF git repo?

ianheggie-oaf committed:
Increase vagrant memory, add swapfile and other work to try and get discourse provisioining

ianheggie-oaf committed:
Deploy from openaustralia repo only and insist on it so commit sha's will be present

Note - this blocks non oaf staff from deploying, which is fine
Also apply rubocop auto formatting
